Improve node tests.

showell opened the pull request.

cPhost commented on frontend_tests/node_tests/general.js:
```diff
@@ -0,0 +1,746 @@
+
+    elem.empty = () => {
+        return elem;
+    };
```
The whole block looks cleaner with:
```javascript
elem.expectOne = () => elem;
elem.removeClass = () => elem;
elem.empty = () => elem;
```

cPhost commented on frontend_tests/node_tests/general.js:
```diff
@@ -0,0 +1,746 @@
+function make_jquery_helper() {
+    const stream_list_filter = jquery_elem();
+    stream_list_filter.is = () => { return true; };
+    stream_list_filter.val = () => { return ''; };
```
Same here, maybe not use return and braces here?
```javascript
stream_list_filter.is = () => true;
stream_list_filter.val = () => '';
```

cPhost commented on frontend_tests/node_tests/general.js:
```diff
@@ -0,0 +1,746 @@
+    const filter = new Filter(filter_terms);
+
+    narrow_state.filter = () => { return filter; };
+    narrow_state.active = () => { return true; };
```
Use this syntax instead:
```javascript
narrow_state.filter = () => filter;
narrow_state.active = () => true;
```

cPhost commented on frontend_tests/node_tests/general.js:
```diff
@@ -0,0 +1,746 @@
+        notifications.window_has_focus = () => { return true; };
+
+        // Make our "test" message appear visible.
+        message_viewport.visible_messages = () => { return test_messages; };
```
Maybe just
```
notifications.window_has_focus = () => true;
message_viewport.visible_messages = () => test_messages;
```

showell commented:
I like this suggestion and honestly didn’t realize es6 supported that.

lonerz commented on frontend_tests/node_tests/popovers.js:
```diff
@@ -0,0 +1,174 @@
+        id: 999,
+        sender_id: alice.user_id,
+    };
+    var row = {};
```
Ah, I guess for future testing this might be used? I don't really see it used other than `rows.id(row)`.

lonerz commented on frontend_tests/node_tests/narrow_activate.js:
```diff
@@ -0,0 +1,201 @@
+        'trigger event',
+    ]);
+
+    channel.post = (opts) => {
```
I'm interested to why this is not earlier because there seems to be another `maybe_report_narrow_time` in the `setTimeout` (L257-260). Seems like that is also run, so we can assert it in the helper above.

lonerz commented:
@showell reviewed all the code!

showell commented:
I addressed all the feedback here.  @lonerz I renamed clear to clear_events in the one place.  For the maybe_report_narrow_time() question, I think it's possible that the cont() logic explains your confusion?

cPhost commented on frontend_tests/node_tests/list_render.js:
```diff
@@ -0,0 +1,156 @@
+(function test_list_render() {
+    const containers = make_containers();
+    const container = containers.container;
+    const parent_container = containers.parent_container;
```
Quick comment on this, you can deconstructing here:
```javascript
const { container, parent_container } =  make_containers();
```

showell commented:
The PR continues to grow. :)

@cPhost Thanks for the suggestion on using the deconstruction syntax--I fixed it.

timabbott commented:
This is great; now that the release is out, merged, thanks @showell!
